llvm/include/llvm/IR/Intrinsics.td

```diff
@@ -1003,6 +1003,14 @@ def int_memset_inline
        NoCapture<ArgIndex<0>>, WriteOnly<ArgIndex<0>>,
        ImmArg<ArgIndex<3>>]>;
 
+// Memset variant that writes a given pattern.
+def int_memset_pattern
+    : Intrinsic<[],
+      [llvm_anyptr_ty, llvm_anyint_ty, llvm_anyint_ty, llvm_i1_ty],
+      [IntrWriteMem, IntrArgMemOnly, IntrWillReturn, IntrNoFree, IntrNoCallback,
```

Comment:
DefaultAttrIntrinsic would hide most of these

Reply:
I dug through this and unfortunately I can't make use of `DefaultAttrIntrinsic` because `nosync` isn't necessarily true. https://reviews.llvm.org/D86021 switched over memset to using DefaultAttrIntrinsic but this was later backed out in a888e492f601644f44b843373039966e7acacd31 due to `nosync` not applying unconditionally.